src/GraphWebhooks/Controllers/WatchController.cs

```diff
@@ -0,0 +1,255 @@
+    /// Implements subscription management endpoints
+    /// </summary>
+    public class WatchController : Controller
+    {
+        private readonly GraphServiceClient _graphClient;
+        private readonly SubscriptionStore _subscriptionStore;
+        private readonly CertificateService _certificateService;
+        private readonly ILogger<WatchController> _logger;
+        private readonly string _notificationHost;
+
+        public WatchController(
+            GraphServiceClient graphClient,
+            SubscriptionStore subscriptionStore,
+            CertificateService certificateService,
+            ILogger<WatchController> logger,
+            IConfiguration config)
+        {
+            _graphClient = graphClient;
+            _subscriptionStore = subscriptionStore;
+            _certificateService = certificateService;
+            _logger = logger;
+
+            _notificationHost = config.GetValue<string>("NotificationHost");
+            if (string.IsNullOrEmpty(_notificationHost) || _notificationHost == "YOUR_NGROK_PROXY")
+            {
+                throw new ArgumentException("You must configure NotificationHost in appsettings.json");
+            }
+        }
+
+        /// <summary>
+        /// GET /watch/delegated
+        /// Creates a new subscription to the authenticated user's inbox and
+        /// displays a page that updates with each received notification
+        /// </summary>
+        /// <returns></returns>
+        [AuthorizeForScopes(ScopeKeySection = "GraphScopes")]
+        public async Task<IActionResult> Delegated()
+        {
+            try
+            {
+                // Delete any existing subscriptions for the user
+                await DeleteAllSubscriptions(false);
+
+                // Get the user's ID and tenant ID from the user's identity
+                string userId = User.FindFirst("http://schemas.microsoft.com/identity/claims/objectidentifier")?.Value;
+                _logger.LogInformation($"Authenticated user ID {userId}");
+                string tenantId = User.FindFirst("http://schemas.microsoft.com/identity/claims/tenantid")?.Value;
+
+                // Get the user from Microsoft Graph
+                var user = await _graphClient.Me
+                    .Request()
+                    .Select(u => new {u.DisplayName, u.Mail, u.UserPrincipalName})
+                    .GetAsync();
+
+                _logger.LogInformation($"Authenticated user: {user.DisplayName} ({user.Mail ?? user.UserPrincipalName})");
+                // Add the user's display name and email address to the user's
+                // identity.
+                User.AddUserGraphInfo(user);
+
+                // Create the subscription
+                var subscription = new Subscription
+                {
+                    ChangeType = "created",
+                    NotificationUrl = $"{_notificationHost}/listen",
+                    Resource = "me/mailfolders/inbox/messages",
```

Comment:
didn't we use to read that from the app settings?

Reply:
Yes, but I've redesigned the app a bit so that it expects specific resource types (message for delegated scenario, chatMessage for app-only). Instead of displaying the raw JSON of each notification payload, I'm now reading deserializing and reading specific fields, then adding that to the table on the watch page.

Reply:
ok, that makes the sample easier to discover/understand for people but a bit less versatile to try against anything. Resovling.